mypyc/emitmodule.py

```diff
@@ -67,6 +68,33 @@
 # A list of (file name, file contents) pairs.
 FileContents = List[Tuple[str, str]]
 
+# The C standard specifies that any number of valid hexadecimal characters are
+# parsed as part of the escaped sequence. Therefore, "\x12345" would be
+# unexpectedly parsed as {0x12345}, instead of {0x123, '4', '5'}. Therefore, we
+# use octal escape sequences which are specified to contain at most three octal
+# digits.
+C_CHAR_MAP = ["\\{:03o}".format(x) for x in range(256)]
+# Most printable characters do not need to be escaped in string literals. We
+# can safely use string.printable here because it always uses the C locale.
+for x in string.printable:
+    C_CHAR_MAP[ord(x)] = x
+# These assignments must be done after string.printable because they are
+# overrides for the printable characters that need to be escaped in string
+# literals.
+C_CHAR_MAP[ord('\'')] = '\\\''
```

Comment:
These are all generated the same way (adding a backslash), so I think it would be an improvement to generate them all mechanically from a list of characters to escape.

Reply:
I considered this, but the neatest way I can think to do this is:

```python
for x in ("'", '"', '\\', 'a', 'b', 'f', 'n', 'r', 't', 'v'):
    escaped = '\\{}'.format(x)
    value = escaped.encode('ascii').decode('unicode_escape')
    C_CHAR_MAP[ord(value)] = escaped
```

I am not sure whether this is an improvement. What do you think?